pkg/asset/machines/master.go

```diff
@@ -81,10 +89,15 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 	mign := &machine.Master{}
 	dependencies.Get(clusterID, installconfig, rhcosImage, mign)
 
-	var err error
-	machines := []machineapi.Machine{}
 	ic := installconfig.Config
+
 	pool := ic.ControlPlane
+	if *pool.Replicas == 0 {
+		return errors.New("at least one master machine must be configured")
```

Comment:
Is this the right place for this?  If we want validation inside `Generate`, why not run the whole `validateControlPlane` on it (we'd have to make that function public)?

Reply:
Ah, I see we had something like this before down below.  Are we concerned about promoting it to a location above the `nonetypes.Name` bail out?

Reply:
https://github.com/openshift/installer/blob/b3b8a8928b1953c00944fae307c933dbcad6cf75/pkg/types/validation/installconfig.go#L161-L163
and
https://github.com/openshift/installer/blob/b3b8a8928b1953c00944fae307c933dbcad6cf75/pkg/types/defaults/installconfig.go#L49-L53
together make it so that controlplane cannot be zero?

So i dropped this check all together. 8967ccd066030ebbb18fb3d19f7dd693df8abc46...c1ae01308bba2c83946b140c3fe5d740ad671a8f   PTAL